test/tstKokkosToolsAnnotations.cpp

```diff
@@ -43,15 +43,15 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(bvh_bvh_allocations_prefixed, DeviceType,
   Kokkos::Tools::Experimental::set_allocate_data_callback(
       [](Kokkos::Profiling::SpaceHandle /*handle*/, char const *label,
          void const * /*ptr*/, uint64_t /*size*/) {
-        BOOST_TEST_MESSAGE(label);
         BOOST_TEST(
             (isPrefixedWith(label, "ArborX::BVH::") || // data member
              isPrefixedWith(label, "ArborX::BVH::BVH::") ||
              isPrefixedWith(label, "ArborX::Sorting::") ||
              isPrefixedWith(label, "Kokkos::SortImpl::BinSortFunctor::") ||
              isPrefixedWith(label,
                             "Kokkos::Serial::") || // unsure what's going on
-             isPrefixedWith(label, "Testing::")));
+             isPrefixedWith(label, "Testing::")),
+            "Failing label is: '" << label << "'");
```

Comment:
These changes were very helpful in seeing what the offending label really was. Printing it instead of the failing condition is much better in my opinion. I could of course push it to a separate pull request if preferred.

Reply:
Preferring the optional failure message to the message is a good idea.
We need to think about the message a bit carefully though.  I deleted all `isPrefixedWith` but the last one (the one with `"Testing::"`) and pasted a sample from the verbose output below
```
<ArborX>/test/tstKokkosToolsAnnotations.cpp:47: error: in "KokkosToolsAnnotations/bvh_bvh_allocations_prefixed<Kokkos__Device<Kokkos__Serial_ Kokkos__HostSpace>>": Failing label is: 'ArborX::BVH::internal_and_leaf_nodes'

<ArborX>/test/tstKokkosToolsAnnotations.cpp:47: info: check 'Failing label is: 'Testing::boxes'' has passed
```
The `info: check 'Failing label is: 'Testing::boxes'' has passed` is a bit silly

Reply:
How about
```diff
diff --git a/test/tstKokkosToolsAnnotations.cpp b/test/tstKokkosToolsAnnotations.cpp
index 8aba028..8299a2c 100644
--- a/test/tstKokkosToolsAnnotations.cpp
+++ b/test/tstKokkosToolsAnnotations.cpp
@@ -14,6 +14,7 @@

 #include <boost/test/unit_test.hpp>

+#include <regex>
 #include <string>

 #include "Search_UnitTestHelpers.hpp"
@@ -43,15 +44,14 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(bvh_bvh_allocations_prefixed, DeviceType,
   Kokkos::Tools::Experimental::set_allocate_data_callback(
       [](Kokkos::Profiling::SpaceHandle /*handle*/, char const *label,
          void const * /*ptr*/, uint64_t /*size*/) {
-        BOOST_TEST_MESSAGE(label);
-        BOOST_TEST(
-            (isPrefixedWith(label, "ArborX::BVH::") || // data member
-             isPrefixedWith(label, "ArborX::BVH::BVH::") ||
-             isPrefixedWith(label, "ArborX::Sorting::") ||
-             isPrefixedWith(label, "Kokkos::SortImpl::BinSortFunctor::") ||
-             isPrefixedWith(label,
-                            "Kokkos::Serial::") || // unsure what's going on
-             isPrefixedWith(label, "Testing::")));
+        std::regex re("^(Testing::"
+                      "|ArborX::BVH::"
+                      "|ArborX::Sorting::"
+                      "|Kokkos::SortImpl::BinSortFunctor::"
+                      "|Kokkos::Serial::" // unsure what's going on
+                      ").*");
+        BOOST_TEST(std::regex_match(label, re),
+                   "determine if \"" << label << "\" matches the regex");
       });

   { // default constructed